### ISSUE TITLE
Allow indexed dll access (or use OleWoo to view existing IDL files)

### ISSUE THREAD

**Issue by Greedquest** [leibnitz27/olewoo#20]
To view the 3rd id in a dll: `C:\Windows\SysWOW64\msvbvm60.dll\3` I had to do `oledump.exe C:\Windows\SysWOW64\msvbvm60.dll\3 > results.idl`. Is there a way to get at different tlib IDs inside olewoo rather than resorting to oledump? It may be that the windows explorer add file dialog doesn't let you enter invalid filepaths like that.

Alternatively a way to view results.idl in OleWoo GUI would be lovely.

Thanks for the tool, really useful!

**Comment by drmason789** [comment]
`olewoo_ui C:\Windows\SysWOW64\msvbvm60.dll\3`

The filepath is passed directly into a Windows API function which I think interprets the \3 as a tlib ID - this is not Explorer functionality, so the file dialog won't deal with it. 

We could possibly take a conventional file path, scan it for tlibs, and present them all in a tabbed format.

**Comment by Greedquest** [comment]
The scanning sounds fantastic - C:\Windows\SysWOW64\msvbvm60.dll\2 is nothing whereas \1 and \3 are. If I don't know in advance these IDs (e.g. when reverse engineering) a function to scan the dll for them would be useful.

**Comment by drmason789** [comment]
If the type library is registered, you could do a search of the registry in HKCR\TypeLib for the dll file path and see which numbers it is suffixed with.

**Comment by Greedquest** [comment]
Thanks for the pointer:), searching HKCR\TypeLib works for `msvbvm60.dll\3` but not `VBE7.dll\3` which I suppose wasn't registered (as in, searching for `VBE7.dll` yields no results with an ID suffix). Would you envisage just looping through `file.dll\1..n` or is there a clever way to extract that info from the dll itself?

**Comment by drmason789** [comment]
I don't know much about the type library APIs, but what I've seen looks a bit crude - type library has to be loaded from a file, so no loading into memory then quickly looping through 1..n, I think it's going to read the while file every time. Probably not a big deal on modern hardware, for n in 1..10.

The type library file format could possibly be used: [http://theircorp.byethost11.com/files/TypeLib.txt](url)

I'm not actually a main contributor - I've just the installer project (msi that installs the UI and cmd line utility) and a project to create a nuget package for oledump, but that PR has been waiting for acceptance for over a year. I could add this indexed access functionality, but I'm not confident it will be merged into the main repository.

**Comment by leibnitz27** [comment]
> I'm not actually a main contributor - I've just the installer project (msi that installs the UI and cmd line utility) and a project to create a nuget package for oledump, but that PR has been waiting for acceptance for over a year. I could add this indexed access functionality, but I'm not confident it will be merged into the main repository.

Oh wow.   Err, sorry.   Cough.

**Comment by leibnitz27** [comment]
This is (unless I'm reading it wrong) more of a limitation of the open file dialog.

You can get around this quickly by just specifying the indexed dll path as a command line arg.

**Comment by Greedquest** [comment]
@leibnitz27 Yeah I didn't realise I could call olewoo.exe with a dll path. Nevertheless, opening different tabs of a dll would be a friendlier UX and a nice expansion - but definitely there's a good workaround that I didn't know about before.

**Comment by drmason789** [comment]
I've submitted a PR for this. It was quite easy once I found out how to do it. I also added the filename with resource name suffix (the \1 or \3 in our example) as the tooltip on the tabs.

I'm wondering if oledump could use this too. e.g. 

`oledump -a msvbvm60.dll`

could dump all the libraries.

Perhaps

`oledump -l msvbvm60.dll`

could list all the libraries, so they could be enumerated in a script.
